File: java/Easy/_171_ExcelSheetColumnNumber.java
package Easy;

import org.testng.annotations.Test;

public class _171_ExcelSheetColumnNumber {
    public int titleToNumber(String columnTitle) {
        char[] arC = columnTitle.toCharArray();
        int res = 0;
        for(int i = arC.length-1, j=0; i>=0; i--, j++){
            int cur = arC[i]-'A' + 1;
            cur*=Math.pow(26, j);
            res += cur;
        }
        return res;
    }

    @Test
    public void test(){
        System.out.println(titleToNumber("ZY"));
    }
}

File: Javascript/_171_ExcelSheetColumnNumber.js
/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function(columnTitle) {
    let chars = columnTitle.split('');
    let res = 0;
    let unit = 0;
    for(let i = chars.length-1; i>=0; i--) {
        res += (chars[i].charCodeAt(0)-64)*Math.pow(26,unit++);
    }
    return res;
};



console.log(titleToNumber("A"));
console.log(titleToNumber("Z"));
console.log(titleToNumber("AA"));
console.log(titleToNumber("AZ"));
console.log(titleToNumber("ZZ"));
console.log(titleToNumber("AAA"));